### PR TITLE
Fix Vue3 unit tests

### PR DESCRIPTION
**Pull request by joseacabaneros**
Fix the latest unit test sheets to pass.
Screenshot 2024-09-16 at 5 37 31 PM

### CODE REVIEW

File: packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview-list.spec.ts [hunk 1]
```diff
@@ -62,13 +59,9 @@ function renderQueryPreviewList({
...
-    getQueryPreviewItemWrappers() {
-      return wrapper.findAllComponents(QueryPreview);
-    },
-    reRender() {
-      // A timeout with no time should resolve after all reactivity + promises involved
-      return new Promise(resolve => setTimeout(resolve));
-    }
+    getQueryPreviewItemWrappers: () => wrapper.findAllComponents(QueryPreview),
+    // A timeout with no time should resolve after all reactivity + promises involved
+    reRender: () => new Promise(resolve => setTimeout(resolve))
```
Comment (victorcg88) [inline packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview-list.spec.ts:64]: Would it not have the same effect as having `await nextTick()`?

File: packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview.spec.ts [hunk 2]
```diff
@@ -1,89 +1,94 @@
...
         queryFeature,
+        maxItemsToRender,
         debounceTimeMs,
         persistInCache
       }
     }
-  ).findComponent(QueryPreview);
+  );
+
+  if (queryPreviewInState) {
```
Comment (victorcg88) [inline packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview.spec.ts:67]: Pretending like I didn't see an `if` in a test 🫣

File: packages/x-components/src/x-modules/related-tags/components/__tests__/related-tag.spec.ts [hunk 3]
```diff
@@ -1,79 +1,64 @@
...
+        props: { relatedTag, highlightCurated: false }
       }
     );
-    const wrapper = wrapperTemplate.findComponent(RelatedTagComponent);
+    const wrapperRelatedTag = wrapper.findComponent(RelatedTag);
 
     return {
       wrapper,
-      wrapperTemplate,
+      wrapperRelatedTag,
```
Comment (victorcg88) [inline packages/x-components/src/x-modules/related-tags/components/__tests__/related-tag.spec.ts:30]:
```suggestion
      wrapperRelatedTag: wrapper.findComponent(RelatedTag);,
```
And delete the const definition above

File: packages/x-components/src/x-modules/facets/components/lists/__tests__/selected-filters.spec.ts [hunk 4]
```diff
@@ -1,56 +1,61 @@
...
+    }
+  );
+  resetXFacetsStateWith(store, facets);
+  nextTick();
 
   const selectedFiltersWrapper = wrapper.findComponent(SelectedFilters);
 
   return {
     wrapper,
     selectedFiltersWrapper,
```
Comment (victorcg88) [inline packages/x-components/src/x-modules/facets/components/lists/__tests__/selected-filters.spec.ts:54]:
```suggestion
    selectedFiltersWrapper: wrapper.findComponent(SelectedFilters);,
```

And remove the above const declaration.

File: packages/x-components/src/x-modules/search/components/__tests__/banners-list.spec.ts [hunk 5]
```diff
@@ -1,98 +1,72 @@
...
+      }
     }
   );
+  resetXSearchStateWith(store, { banners });
+  await nextTick();
 
   const bannersListWrapper = wrapper.findComponent(BannersList);
 
   return {
     wrapper: bannersListWrapper,
```
Comment (victorcg88) [inline packages/x-components/src/x-modules/search/components/__tests__/banners-list.spec.ts:40]:
```suggestion
    wrapper: bannersListWrapper: wrapper.findComponent(BannersList);,
```

And delete the const definition above

File: packages/x-components/src/x-modules/search/components/__tests__/promoteds-list.spec.ts [hunk 6]
```diff
@@ -1,98 +1,71 @@
...
+      }
     }
   );
+  resetXSearchStateWith(store, { promoteds });
+  await nextTick();
 
   const promotedsListWrapper = wrapper.findComponent(PromotedsList);
 
   return {
     wrapper: promotedsListWrapper,
```
Comment (victorcg88) [inline packages/x-components/src/x-modules/search/components/__tests__/promoteds-list.spec.ts:39]:
```suggestion
    wrapper: promotedsListWrapper: wrapper.findComponent(PromotedsList);,
```

File: packages/x-components/src/x-modules/facets/components/lists/__tests__/selected-filters.spec.ts [hunk 7]
```diff
@@ -1,56 +1,61 @@
...
+    {
+      store,
+      global: { plugins: [installNewXPlugin({ store, initialXModules: [facetsXModule] })] },
+      data: () => {
+        return { facetsIds };
+      }
+    }
+  );
+  resetXFacetsStateWith(store, facets);
+  nextTick();
```
[the remark this answers is not in the document]
Comment (joseacabaneros) [reply packages/x-components/src/x-modules/facets/components/lists/__tests__/selected-filters.spec.ts:48]: Yes, added

File: packages/x-components/src/x-modules/facets/components/lists/__tests__/selected-filters.spec.ts [hunk 8]
```diff
@@ -1,56 +1,61 @@
...
-    data: () => ({ facetsIds })
-  });
+  const store = new Store<DeepPartial<RootXStoreState>>({});
+  const wrapper = mount(
+    {
+      components: { SelectedFilters },
+      template
+    },
+    {
+      store,
```
[the remark this answers is not in the document]
Comment (joseacabaneros) [reply packages/x-components/src/x-modules/facets/components/lists/__tests__/selected-filters.spec.ts:40]: It isn't necessary. Only needed for the XPlugin

File: packages/x-components/src/x-modules/facets/components/lists/__tests__/selected-filters.spec.ts (hunk 4 above)
Comment (joseacabaneros) [reply packages/x-components/src/x-modules/facets/components/lists/__tests__/selected-filters.spec.ts:54]: Done

File: packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts [hunk 9]
```diff
@@ -1,246 +1,185 @@
...
 
+async function render({
+  nextQueries = getNextQueriesStub(),
+  template = '<NextQueries :suggestions="suggestions" :highlightCurated="highlightCurated"/>'
+} = {}) {
+  const store = new Store<DeepPartial<RootXStoreState>>({});
+  const resetStoreState = async (state: Parameters<typeof resetXNextQueriesStateWith>[1]) => {
+    resetXNextQueriesStateWith(store, state);
+    await nextTick();
+  };
```
[the remark this answers is not in the document]
Comment (joseacabaneros) [reply packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts:22]: This reset function is used further forward in two more tests to set the next queries to test.
If we get rid of this util function, we would return the `store` in the `render` function to do it by hand.
I prefer to have it encapsulated in a util function

File: packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview-list.spec.ts (hunk 1 above)
Comment (joseacabaneros) [reply packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview-list.spec.ts:64]: `+ promises involved` is the key. I replaced this function by `flushPromises` which in essence is the same we are doing in the `reRender`

File: packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview.spec.ts (hunk 2 above)
Comment (joseacabaneros) [reply packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview.spec.ts:67]: 😅 I could remove it, making `queryPreviewInState` param required and passing the state when it is needed

File: packages/x-components/src/x-modules/related-tags/components/__tests__/related-tag.spec.ts (hunk 3 above)
Comment (joseacabaneros) [reply packages/x-components/src/x-modules/related-tags/components/__tests__/related-tag.spec.ts:30]: Done

File: packages/x-components/src/x-modules/search/components/__tests__/banners-list.spec.ts (hunk 5 above)
Comment (joseacabaneros) [reply packages/x-components/src/x-modules/search/components/__tests__/banners-list.spec.ts:40]: done

File: packages/x-components/src/x-modules/search/components/__tests__/promoteds-list.spec.ts (hunk 6 above)
Comment (joseacabaneros) [reply packages/x-components/src/x-modules/search/components/__tests__/promoteds-list.spec.ts:39]: Done

File: packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (victorcg88) [reply packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts:22]: Perfect, I didn't know it was used later in the tests.

**Review by victorcg88 — APPROVED** [review]

**Pull request merged**